gateway/buckets.go

```diff
@@ -84,9 +83,12 @@ func (g *Gateway) renderBucketPath(c *gin.Context, ctx context.Context, threadID
 		return
 	}
 	if !rep.Item.IsDir {
-		if err := g.buckets.PullPath(ctx, buck.Key, pth, c.Writer); err != nil {
+		contentType, r, err := getContentTypeFromPullPath(ctx, g.buckets, buck.Key, pth)
+		if err != nil {
 			render404(c)
```

Comment:
I could be reading the diff wrong, but `return` needed after `render404`?

Reply:
Oh yeah, I'll add that and merge 👍🏼